File: psra/service/src/main/scala/by/arhor/psra/service/PhotoService.scala
package by.arhor.psra.service

import java.util

import by.arhor.psra.dto.{CommentDto, PhotoDto, UserDto}
import by.arhor.psra.model.Photo

trait PhotoService extends Service[Photo, PhotoDto, String] {

  def findPhotosByTag(tag: String, requester: UserDto): util.List[PhotoDto]

  def findCommentsByPhotoId(pid: String): util.List[CommentDto]

  def addCommentToPhoto(photoId: String, username: String, dto: CommentDto)

}

File: psra/service/src/main/java/by/arhor/psra/service/PhotoService.java
package by.arhor.psra.service;

import by.arhor.psra.dto.CommentDto;
import by.arhor.psra.dto.PhotoDto;

import java.util.List;

public interface PhotoService
    extends Service<PhotoDto, String>
          , CanCreate<PhotoDto>
          , CanUpdate<PhotoDto> {

  List<PhotoDto> findPhotosByTag(String tag, String username);

  List<CommentDto> findCommentsByPhotoId(String pid);

  CommentDto addCommentToPhoto(String photoId, String username, CommentDto dto);

}
